pkg/manager/member/common_store_failover.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+// StoreAccess contains the common set of functions to access the properties of TiKV and TiFlash types
+type StoreAccess interface {
+	GetFailoverPeriod(cliConfig *controller.CLIConfig) time.Duration
+	GetMemberType() v1alpha1.MemberType
+	GetMaxFailoverCount(tc *v1alpha1.TidbCluster) *int32
+	GetStores(tc *v1alpha1.TidbCluster) map[string]v1alpha1.TiKVStore
+	SetFailoverUIDIfAbsent(tc *v1alpha1.TidbCluster)
+	CreateFailureStoresIfAbsent(tc *v1alpha1.TidbCluster)
+	GetFailureStores(tc *v1alpha1.TidbCluster) map[string]v1alpha1.TiKVFailureStore
+	SetFailureStore(tc *v1alpha1.TidbCluster, storeID string, failureStore v1alpha1.TiKVFailureStore)
+	ClearFailStatus(tc *v1alpha1.TidbCluster)
+	GetStsDesiredOrdinals(tc *v1alpha1.TidbCluster, excludeFailover bool) sets.Int32
+}
+
+// commonStoreFailover handles the common failover logic for TiKV and TiFlash
+type commonStoreFailover struct {
+	deps        *controller.Dependencies
+	storeAccess StoreAccess
+}
+
```

Comment:
Explicitly enforce static check that struct `commonStoreFailover` implements `Failover` interface.
```suggestion

var _ Failover = (*commonStoreFailover)(nil)

```